src/Microsoft.ML.Experimental/Microsoft.ML.Experimental.csproj

```diff
@@ -5,6 +5,10 @@
     <IncludeInPackage>Microsoft.ML.Experimental</IncludeInPackage>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="System.Data.SqlClient" Version="4.6.1" />
```

Comment:
We shouldn't be adding this dependency to the core logic. If we have APIs that are dependent on `System.Data.SqlClient`, those need to be split out into a separate nuget package.

Reply:
This is just in the experimental package. I don't feel its appropriate to have 3-4 different experimental packages (one for each database type) and we should probably split these out when they go into the main branch.

Reply:
Just know that is undone work that will have to be done at some time.

Reply:
Thinking about it, I believe if we use the DbProviderFactory approach suggested above, we probably won’t need a specific reference to the SQL client. Instead the user would pass us the SqlClientProviderFactory.

Reply:
@eerhardt, @CESARDELATORRE. How are we going to handle resource cleanup here? `DbProviderFactory` doesn't expose IDisposable. So are we just going to track the input and call `DbProviderFactory.CreateCommand()` and then dispose it always?

Reply:
Also, how is it expected to work with users needing to specify a single `commandText` for all commands in something like the simple Sdca test in the PR?

If users are expected to follow a pattern like https://github.com/dotnet/machinelearning/pull/4050#issuecomment-517451687, then they will still need to manually track all of the state around the created connections/commands and shared state (like `commandText`, etc).

Reply:
That would be my suggestion. It is how typical ADO.NET code is written.

Check out:

https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/dbconnection-dbcommand-and-dbexception#retrieving-data-example

and the older article:
https://docs.microsoft.com/en-us/previous-versions/dotnet/articles/ms971499(v=msdn.10)

Reply:
Btw - you don't dispose the Factory.  You dispose the Connection, Command, and DataReader.